app/components/post_textbox/post_textbox.js

```diff
@@ -175,8 +185,22 @@ export default class PostTextbox extends PureComponent {
         }
     };
 
-    handleAndroidKeyboard = () => {
-        this.blur();
+    handleHardwareEnterKey = () => {
+        if (this.state.keyboardState === 'hidden') {
```

Comment:
we shouldn't care about the state of the software keyboard if I'm using an external keyboard and press Shift+Enter to post the message

Reply:
on the other hand there is the need to use the `hidden` state to trigger the below function as this will also apply to pressing the enter key in the soft keyboard

Reply:
Yes this is actually necessary just like what you said. When tapping the enter button even on the soft-keyboard, this will get triggered and it won't possible do a line break.